### ISSUE TITLE
User Guide: Clarification for the intent for generating resources as go structs instead of yaml

### ISSUE THREAD

**Issue by benjaminapetersen** [operator-framework/operator-sdk#416]
Still looking at the `Quickstart` & wanting to make sure the intent is clear.  It's interesting this generates a `handler.go` file, that has: 

`func newbusyBoxPod(cr *v1alpha1.NodejsAuth) *corev1.Pod {`
… and 30 lines of `go` to basically create a `pod` manifest. 

Is this the pattern users should take for generating all of the resources?  For example, say I am managing an app that has a couple `deployments`, `secrets`, a `statefulset`, etc.  I would build these manifests out in `go`, rather than creating `yaml` files && reading them?

On the one hand, that seems handy, for shared things like `labels`, or `service names`.  Generally there is a lot of duplicate information across `yaml` files that is a hassle to maintain over time.  Helm solves this problem with a `values.yaml` and generates manifests from templates. The `operator-sdk` implies your first step is to translate it all into `go` functions like `newbusyBoxPod`.

If correct, great.  If not, would love to see further guidance on this topic.  

Thanks!

**Comment by hasbro17** [comment]
> Is this the pattern users should take for generating all of the resources?

Yes, the operator can just define all the resources(Custom Resource, Pods, Service etc) as go structs and then pass them on to the dynamic client to create them.

Passing the YAML manifests inside the operator, reading and decoding the YAML into runtime.Objects and then passing them to the client would be a very roundabout(and more complicated) way to do the same thing.

> The operator-sdk implies your first step is to translate it all into go functions like newbusyBoxPod

Not sure about the translate part, but since with the SDK the operator in written in Go then yes it's expected that you will use Go types for all the various resources that the handler deals with.

We do point out the [reference memcached handler](https://github.com/operator-framework/operator-sdk/blob/master/example/memcached-operator/handler.go.tmpl) in the [user-guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#define-the-handler)
And we've also linked to the [SDK samples repo](https://github.com/operator-framework/operator-sdk#samples) on the README that gives a few examples on how handlers are supposed to be written.

If the documentation around any of this is lacking(or not highlighted clearly enough) then feel free to suggest changes. Or close the issue if the above answered your question.

**Comment by benjaminapetersen** [comment]
@hasbro17 thanks! Thats helpful. 

I think examples are fantastic following a thorough walkthrough that clearly communicates intent.

I opened #417 to suggest improving what a reconciliation loop should look like (per the comment that essentially says "don't make yours look like this").

I opened #422 to suggest a step-by-step guide for writing handlers, ensuring a user can appreciate what this component brings to the table.

I just updated the title of this issue to improve specificity.  I think "yaml manifests are no longer important" is probably something that should be clear somewhere in the guide?  For example, even looking at the [helm-app-operator-kit](https://github.com/operator-framework/helm-app-operator-kit) gives the impression that some kind of manifest file is still important.  I now see this was a mistake on my part.  

Thanks!
